### PR TITLE
fix: use config entry runtime_data to get `zwave_js` client

### LINKED ISSUE #369: ISSUE: Locks Not Updating
**Describe the bug**
No lock actions are being performed. Codes stuck as adding, deleting. I just noticed this after the update to Home Assistant update to 2024.6.0

**Environment (please complete the following information):**

- OS: [e.g. HassOS/Raspbian/CentOS]
- Type of system that HA is running on: [e.g. Nuc]
- Home Assistant version: 2024.6.0
- Hassio/Docker/Core?
- Component version: 
- Z-Wave integration name: zwave_js UI
- Lock make and model: Yale Assure 2

**Logs**
2024-06-06 10:04:54.116 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:04:55.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:04:59.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:00.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:04.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:04.386 DEBUG (MainThread) [custom_components.keymaster.binary_sensor] Can't access Z-Wave JS data client.
2024-06-06 10:05:05.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:06.393 DEBUG (MainThread) [custom_components.keymaster.binary_sensor] Can't access Z-Wave JS data client.
2024-06-06 10:05:09.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:10.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:14.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:15.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:19.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:20.691 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:24.116 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:25.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:29.116 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:30.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:34.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:34.386 DEBUG (MainThread) [custom_components.keymaster.binary_sensor] Can't access Z-Wave JS data client.
2024-06-06 10:05:35.692 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:36.394 DEBUG (MainThread) [custom_components.keymaster.binary_sensor] Can't access Z-Wave JS data client.
2024-06-06 10:05:39.116 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:40.691 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)
2024-06-06 10:05:44.115 DEBUG (MainThread) [custom_components.keymaster] Finished fetching keymaster data in 0.000 seconds (success: True)

**Screenshots**
image

**Additional context**

### PR DESCRIPTION
**Pull request by Snuffy2**
## Proposed change
100% of the credit goes to @bikemike

Changes to use config entry runtime_data for zwave.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #369
- This PR is related to issue:

### CODE REVIEW

File: custom_components/keymaster/helpers.py [hunk 1]
```diff
@@ -147,7 +147,12 @@ async def async_update_zwave_js_nodes_and_devices(
     child_locks: List[KeymasterLock],
 ) -> None:
     """Update Z-Wave JS nodes and devices."""
-    client = hass.data[ZWAVE_JS_DOMAIN][entry_id][ZWAVE_JS_DATA_CLIENT]
+    try:
+        zwave_entries = hass.config_entries.async_entries(ZWAVE_JS_DOMAIN)
+        zwave_entry = zwave_entries[0] if zwave_entries else None
```
Comment (firstof9) [inline custom_components/keymaster/helpers.py]: I think we may need to loop and check the states to look for `ConfigEntryState.LOADED` and use the loaded entry.

**Review by firstof9 — CHANGES_REQUESTED** [review]

File: custom_components/keymaster/helpers.py (hunk 1 above)
Comment (firstof9) [reply custom_components/keymaster/helpers.py]: Maybe something like:

```python
zwave_entry = entry for entry in zwave_entries if entry.state == ConfigEntryState.LOADED
```

I'm not 100% sure this code is usable but you get the idea.

Comment (tykeal) [reply custom_components/keymaster/helpers.py]:
```python
zwave_loaded_entries = [entry for entry in zwave_entries if entry.state == ConfigEntryState.LOADED]
zwave_entry = zwave_loaded_entries[0] if zwave_loaded_entries else None
```

This is what it needs to change to as the ```x for x in y``` syntax is a list comprehension (i tried it outside of a list and python did not like it :laughing: )

Also

```python
from homeassistant.config_entries import ConfigEntryState
```
needs to be added to the imports.

File: custom_components/keymaster/binary_sensor.py [hunk 2]
```diff
@@ -131,10 +131,10 @@ async def async_update(self) -> None:
                 self._lock_found = True
 
         try:
-            client = self.hass.data[ZWAVE_JS_DOMAIN][self.lock_config_entry_id][
-                ZWAVE_JS_DATA_CLIENT
-            ]
-        except KeyError:
+            zwave_entries = self.hass.config_entries.async_entries(ZWAVE_JS_DOMAIN)
+            zwave_entry = zwave_entries[0] if zwave_entries else None
```
Comment (tykeal) [inline custom_components/keymaster/binary_sensor.py]: See comments against the ```helpers.py``` file as it is needed here as well

**Review by tykeal — CHANGES_REQUESTED** [review]
Needs a bit more work after some testing on one of my HA systems

**Comment by firstof9** [comment]
One last request would be to format using black.
Should be good after that. Thanks @Snuffy2

**Comment by Snuffy2** [comment]
Gladly, I actually had my auto-formatting turned off and didn't notice.

**Comment by firstof9** [comment]
@Snuffy2 you tested the changes locally as well ya?

**Comment by Snuffy2** [comment]
Yes, it is running correctly on my machine without errors.

**Pull request merged**